### PR TITLE
[DOPS-1165] add resource mode

### PR DESCRIPTION
**Pull request by volkorny**

### CODE REVIEW

File: pkg/selector.go [hunk 1]
```diff
@@ -51,8 +68,46 @@ func (s *Selector) claimEndpoint(ctx context.Context, endpoint *registry.Endpoin
...
+					log.Infof("Resource on DNSimple: '%s'", registryRecord.Resource)
+					log.Infof("Resource on K8S: '%s'", endpoint.Resource)
+
+					if registryRecord.Resource == endpoint.Resource {
+						log.Debugf("Resource info up to date for '%s'", registryRecord)
+						continue
+					}
+
+					log.Infof("Updating Resource info for '%s' to '%s'", registryRecord, endpoint.Resource)
+					updatedRecord := registryRecord.NewRecord(s.cfg.CurrentOwnerID, endpoint.Resource)
```
Comment (volodymyr-mykhailyk) [inline pkg/selector.go]: when updating resource - we should use owner from the endpoint, not config.

File: provider/dnsimple/dnsimple.go [hunk 2]
```diff
@@ -52,11 +52,12 @@ func NewDnsimpleProvider(cfg *pkg.Config, zones []string) (provider.Provider, er
...
-	if err != nil {
-		return nil, err
-	}
-	providerInstance.accountID = int64ToString(whoamiResponse.Data.Account.ID)
+	// whoamiResponse, err := providerInstance.identity.Whoami(context.Background())
+	// if err != nil {
+	// 	return nil, err
+	// }
+	// providerInstance.accountID = int64ToString(whoamiResponse.Data.Account.ID)
+	providerInstance.accountID = "77458"
```
Comment (volodymyr-mykhailyk) [inline provider/dnsimple/dnsimple.go]: We should not hardcode account id

File: source/istio_virtualservice.go [hunk 3]
```diff
@@ -186,6 +186,9 @@ func (sc *virtualServiceSource) endpointsFromVirtualService(ctx context.Context,
 		hosts = append(hosts, host)
 	}
 
+	// hosts contains dublicates here
```
Comment (volodymyr-mykhailyk) [inline source/istio_virtualservice.go]: Add TODO string if the comment identifying things we should investigate

File: pkg/selector.go [hunk 4]
```diff
@@ -51,8 +68,46 @@ func (s *Selector) claimEndpoint(ctx context.Context, endpoint *registry.Endpoin
...
+func (s *Selector) claimEndpointResource(ctx context.Context, endpoint *registry.Endpoint, zone *registry.Zone) (updatedRecords int, err error) {
+	hostDiscovered := false
+	for _, host := range zone.Hosts {
+		if endpoint.Host == host.Name {
+			log.Debugf("Host record found for '%s'", endpoint)
+			hostDiscovered = true
+			if host.IsManaged() {
+				for _, registryRecord := range host.RegistryRecords {
+
+					log.Infof("Resource on DNSimple: '%s'", registryRecord.Resource)
```
Comment (volodymyr-mykhailyk) [inline pkg/selector.go]: These should be debug statements

**Review by volodymyr-mykhailyk — APPROVED** [review]
Looks good in general. Please revert account id, and fix other minor comments. We can merge and then redesign operating modes and cli arguments in separate PR

File: provider/dnsimple/dnsimple.go (hunk 2 above)
Comment (volkorny) [reply provider/dnsimple/dnsimple.go]: done

File: source/istio_virtualservice.go (hunk 3 above)
Comment (volkorny) [reply source/istio_virtualservice.go]: done

File: pkg/selector.go (hunk 4 above)
Comment (volkorny) [reply pkg/selector.go]: done

**Pull request merged**